pkg/converters/ingress/annotations/backend.go

```diff
@@ -727,23 +727,31 @@ func (c *updater) buildBackendTimeout(d *backData) {
 func (c *updater) buildBackendWAF(d *backData) {
 	config := d.mapper.GetBackendConfig(
 		d.backend,
-		[]string{ingtypes.BackWAF},
+		[]string{ingtypes.BackWAF, ingtypes.BackWAFMode},
 		func(path *hatypes.BackendPath, values map[string]*ConfigValue) map[string]*ConfigValue {
-			waf, found := values[ingtypes.BackWAF]
-			if !found {
+			waf, foundWAF := values[ingtypes.BackWAF]
+			if !foundWAF {
 				return nil
 			}
 			if waf.Value != "modsecurity" {
-				c.logger.Warn("ignoring invalid WAF mode on %s: %s", waf.Source, waf.Value)
+				c.logger.Warn("ignoring invalid WAF module on %s: %s", waf.Source, waf.Value)
 				return nil
 			}
 			return values
 		},
 	)
 	for _, cfg := range config {
-		d.backend.WAF = append(d.backend.WAF, &hatypes.BackendConfigStr{
-			Paths:  cfg.Paths,
-			Config: cfg.Get(ingtypes.BackWAF).Value,
+		wafModule := cfg.Get(ingtypes.BackWAF).Value
+		wafMode := cfg.Get(ingtypes.BackWAFMode).Value
+		if wafMode != "On" && wafMode != "DetectOnly" {
+			wafMode = "On"
+		}
```

Comment:
* What if mode was `deny` and `detect` in lowercase? This could be used in future improvements where `On` could lead to ambiguity. This would also change the doc and the PR description.
* Add a warning log if mode doesn't match, something like `invalid '%s' on %s, using 'on' instead`, so the admin can have a hint if his proxy is misbehaving.
* Add the default value to `default.go` so this will be documented in code in one single place.

Reply:
:heavy_check_mark: